File: leetcode/python/1680-concatenationOfConsecutiveBinaryNumbers.py
class Solution:
    def countBits(self, n: int) -> int:
        cnt = 0
        while n > 0:
            cnt += 1
            n >>= 1
        return cnt 
    
    def concatenatedBinary(self, n: int) -> int:
        MOD = (10 ** 9) + 7
        res, nBits = 0, 0
        for i in range(1, n + 1):
            nBits = self.countBits(i)
            res = (res << nBits) | i
            res %= MOD
        return res
    
def main():
    s = Solution()
    number = 3
    print(s.concatenatedBinary(number))

if __name__ == "__main__":
    main()
    

File: leetcode/go/1680-concatenationOfConsecutiveBinaryNumbers.go
//go:build ignore

package main

import "fmt"

func countBits(n int) int {
	count := 0
	for n > 0 {
		count++
		n >>= 1
	}
	return count
}

func concatenatedBinary(n int) int {
	res := 0
    MOD := 1000000007
	for i := 1; i <= n; i++ {
		res = (res << countBits(i)) | i
		res = res % MOD
	}
	return res
}

func main() {
	number := 3
	fmt.Print(concatenatedBinary(number))
}